tests/test_esss_fix_format.py

```diff
@@ -783,3 +783,20 @@ def test_exclude_patterns_relative_path_fix(tmp_path, monkeypatch):
     include_patterns = ['*.py']
     exclude_patterns = cli.read_exclude_patterns(config_file)
     assert not cli.should_format('drafts/foo.py', include_patterns, exclude_patterns)[0]
+
+
+def test_utf8_error_handling(tmpdir):
+    file_with_no_uft8 = tmpdir.join("test.cpp")
+    file_with_no_uft8.write('''é'''.encode('UTF-16'), 'wb')
+
+    file_with_uft8 = tmpdir.join("test2.cpp")
+    file_with_uft8.write('''e''', 'w')
```

Comment:
I would suggest (no require) `file_with_no_uft8.write('''é'''.encode('UTF-8'), 'wb') ` Other wise it is file with an utf-8 BOM and plain ASCII contents.

Reply:
Maybe you need for that `'UTF-8-SIG'`